### ISSUE TITLE
Writing Console Applications Article

### ISSUE THREAD

**Issue by cartermp** [dotnet/core-docs#180]
The guide will cover how to use the new cross-platform tooling to build a new console application.

I propose two sections:

1. An incremental guide starting at a basis for compilation, eventually showcasing how you can tie multiple projects together (and test test them!).

2. A fun case study: writing a web scraper which grabs the titles and synopses of each episode in a given [Game of Thrones Season](http://www.imdb.com/title/tt0944947/episodes?season=1) from IMDB.  The output would look something like this:

   ```
   Game of Thrones: Season 1
   -------------------------
   Episode 1: Winter is Coming
   Synopsis: Jon Arryn, the Hand of the King, is dead. King Robert Baratheon plans to ask his oldest friend, Eddard Stark, to take Jon's place. Across the sea, Viserys Targaryen plans to wed his sister to a nomadic warlord in exchange for an army.

   Episode 2: The Kingsroad
   Synopsis: ...
   ...
   ```

   It would showcase a few libraries (like `System.Net.Http` and a`System.Text.RegularExpressions`), and probably link to the async docs for people interested in that.

The code used here would all live in the `/samples` directory and be embedded directly into the document.

**Comment by blackdwarf** [comment]
@cartermp 

Like the idea. However, there are few problems with what you proposed:

1. The first foray into the console applications should not contain info on managing multiple projects and the like. It should link to these topics further in the documentation. When I'm a new user, I want to get my feet wet without drinking concepts from a firehose.
2. I'm not sure IMDB allows HTML scraping in their Terms of Use. They do have an API, I think, so that would be good to use.
3. If you **must** do HTML scraping, choose another site for it. 
4. If you **must** do HTML scraping, please don't use RegEx-es and the like, but a proper HTML parser. This is what is known as an "oppinionated problem" in app development and we don't want people to point to docs and say "you guys are doing it wrong". Especially [this guy](http://blog.codinghorror.com/parsing-html-the-cthulhu-way/). 
5. We should really have the app be more parametrized, correct? So, for instance, a console client for IMDB in general would be much better than a thing that does just GoT. 

So, let's just get some other app built, eh?

**Comment by mairaw** [comment]
I'd avoid using something like GoT in our content since I believe is trademarked.

**Comment by blackdwarf** [comment]
@mairaw good point. :) Though, in this case, you are "just" using it from a 3rd party provider, so the real problem would be stuff like IMDB terms of use (they might not give you the right to use their data wherever and however you want).

**Comment by mairaw** [comment]
Agreed. But I'd say not only what the IMDB terms say but what our legal team says of using another site in our samples. Even names used in sites/samples/etc. have to come from a fictitious name database. I'm just saying that if it's not something generic where you provide an URL as a parameter and do something with it, I'd make sure it's okay to use it.

**Comment by cartermp** [comment]
Good points @blackdwarf and @mairaw.  Perhaps something like [fetching XKCD transcripts and alt texts](http://xkcd.com/1/info.0.json)?  All his info is licensed under Creative Commons so I think any legal issues would be minimal at most.

The main goal for the second one is to be kind of fun and showcase something more than the typical, sorta boring pieces of sample code that we usually supply (like fetching string values from dotnetfoundation.org or something).  I'm open to suggestions if you have a good idea here.

@blackdwarf regarding the first section:

I should clarify, this article would sit under the Console Applications part of the TOC.  I'm not intending for this to be [the article in Getting Started](https://github.com/dotnet/core-docs/blob/master/docs/getting-started/first-console-app.md) which covers the "create something super basic" scenario.

You don't think that an incremental guide would suit well here?  I don't think it's drinking from a firehose so long as it's well-paced.  I was imagining something like this:

1. Using `dotnet init`, `dotnet compile`, and `dotnet run` to run the Hello World
2. Examining the components which allowed for the above to happen
3. Using `dotnet clean` to clear our the `bin` folder and `dotnet compile --native` to show how to create a single binary, and then how to run that binary
4. Extending it to create a new type in another file (like Dog.cs), instantiate that new type, and print out some of the values.  `dotnet clean` and `dotnet compile --native` would be shown again
5. Extending it to show how you can incorporate files from other folders by moving the new type into a `Models` folder.
6. Moving the binary out of the `bin` folder and executing it somewhere else

Or perhaps breaking this up so that it fits in with the TOC as-is?

Overview and Moving Pieces --> parts 1 and 2 from above
Console applications paradigm --> other parts not applicable?
Creating single binaries with native compilation --> parts 3 and 6

I think that the 1-6 points could fit under the "Console applications paradigm" section if that were used to incrementally build up someone's understanding of the paradigm with a growing example ... but at the same time, I can see how it would be awkward to effectively copy things like creating binaries with native compilation across different articles.

Personally I'd like to see some kind of incremental guide in one of the sections.  I think "Console applications paradigm" is a decent candidate there, but one of the other sections might be better.

What do you think?

**Comment by blackdwarf** [comment]
@cartermp somehow I read that this is supposed to be the console application article in the "Getting started" portion of the docs. It is a fine approach for the main parts. As far as *how* that would be done, I'm completely open to adding a couple of more documents to the TOC; no need to remain fixed at these. The "Console paradigm" was supposed to be an article that introduces and explains concepts like the shared runtime, self-contained runtime, single native binaries etc. I believe these are important to introduce to new people (not that they are unique, for instance python has the same set of concepts). So you could just add a couple of more articles to the TOC and write this out. 

As far as a "cool example" goes, if you do want to keep the HTML parsing stuff, XKCD is fine for that, but please don't do regexes for all the reasons I mentioned.

**Comment by cartermp** [comment]
@blackdwarf gotcha!  Thinking about it, I think I'll focus just on adding an article or two to the TOC which are basically guides (the two scenarios I described).  The other parts seem like they'd be better written by you, especially since you drove the console paradigm :).

**Comment by BillWagner** [comment]
This is partially addressed in PR #190 That's still open, and needs an update to the code. It's using the older CLI syntax. 

There's also no content coming for http://dotnet.github.io/docs/getting-started/first-console-app.html I could not find an issue that's tracking that, so I wrote #466 for that article and self-assigned it.

**Comment by mairaw** [comment]
This is done
